bokeh/charts/_data_adapter.py

```diff
@@ -137,6 +138,13 @@ def validate_values(values):
 
             return values
 
+        elif hasattr(values, '__array__'):
```

Comment:
Yeah, it looks like the errors you were encountering were on the Blaze side and probably up to me to resolve them.  I suggest that Bokeh not try to work around these, but instead assume that `asarray` works as expected.